unit_tests/src/test_atomic_apply_acl_operations.cpp

```diff
@@ -636,6 +636,109 @@ TEST_CASE("#7408: Atomic acls respect entity zone")
             }) != e);
 }
 
+TEST_CASE("#7966: Atomic acls accept all acls")
+{
+    using namespace std::string_literals;
+
+    rodsEnv env;
+    _getRodsEnv(env);
+    irods::experimental::client_connection conn;
+    const auto* const user_home = env.rodsHome;
+    const auto* const acl_user_name = "atomic_acl_user";
+
+    ua::user acl_user{acl_user_name};
+
+    // clean up users after end
+    irods::at_scope_exit remove_users{[&] {
+            CHECK_NOTHROW(ua::client::remove_user(conn, acl_user));
+    }};
+
+    REQUIRE_NOTHROW(ua::client::add_user(conn, acl_user));
+
+    REQUIRE(ua::client::exists(conn, acl_user));
+
+    // clang-format off
+
+    const std::array new_acls_and_perms{ std::make_pair("read_metadata", fs::perms::read_metadata), 
+                                         std::make_pair("read_object", fs::perms::read),
+                                         std::make_pair("create_metadata", fs::perms::create_metadata),
+                                         std::make_pair("modify_metadata", fs::perms::modify_metadata),
+                                         std::make_pair("delete_metadata", fs::perms::delete_metadata),
+                                         std::make_pair("create_object", fs::perms::create_object),
+                                         std::make_pair("modify_object", fs::perms::write),
+                                         std::make_pair("delete_object", fs::perms::delete_object) };
```

Comment:
Is the reason that we are only testing the "new" permissions because the others (null, read, write, and own) are being tested elsewhere? Should we include them here anyway, if only to avoid future confusion?

Reply:
yes please, just loop all of them.   could even 'replace' the older test if that helps clarify.